src/rules/whitespaceRule.ts

```diff
@@ -122,6 +128,21 @@ function walk(ctx: Lint.WalkContext<Options>) {
                 if (options.preblock) {
                     checkForTrailingWhitespace(node.getFullStart());
                 }
+                if (options.postblockPrekeyword) {
+                    const nextToken = utils.getNextToken(node as ts.Block)!;
+                    if (!utils.isSameLine(sourceFile, node.end, nextToken.getStart())) {
+                        break;
+                    }
+                    switch (nextToken.kind) {
+                        case ts.SyntaxKind.WhileKeyword:
+                        case ts.SyntaxKind.ElseKeyword:
+                        case ts.SyntaxKind.CatchKeyword:
+                        case ts.SyntaxKind.FinallyKeyword:
+                            if (node.end === nextToken.getStart()) {
```

Comment:
you can simply use `checkForTrailingWhitespace(node.end)`

Reply:
Changed as you said.